AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin/AWSDataStorePlugin.swift

```diff
@@ -69,6 +69,13 @@ final public class AWSDataStorePlugin: DataStoreCategoryPlugin {
             iStorageEngineSink = newValue
         }
     }
+    
+    var isEagerLoad: Bool {
```

Comment:
we could alternatively do this at config time, `isEagerLoad = !(ModelRegistry.models.contains { $0.rootPath != nil })`

Reply:
code has been updated to remove the public interface that was useful during testing. models generated with the LazyReference will always be lazy loaded when using DataStore, ie. the `isEagerLoaded` is set to false. There is one case that doesn't apply, that is for DataStore.save() where the nested model is required for syncing to cloud, thus the associated model is eager loaded on save